services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java

```diff
@@ -1063,4 +942,41 @@ public int getPort()
       return port;
     }
   }
+  private ArgumentCaptor<Exception> getArgumentCaptor(ServerConfig serverConfig, String errorMessage, boolean haverequest) throws IOException
```

Comment:
I like the idea of returning an `ArgumentCaptor` here rather than doing all the assertions in the private method. I would still prefer passing the lambda to denote which method of `AsyncQueryForwardingServlet ` is being invoked. It would essentially look similar to the JUnit `Assert.assertThrows`.